private/model/api/example.go

```diff
@@ -0,0 +1,379 @@
+	}
+
+	return strings.TrimSpace(buf.String())
+}
+
+func generateExampleInput(ex Example) string {
+	if ex.Operation.HasInput() {
+		return buildShape(&ex.Operation.InputRef, ex.Input, false)
+	}
+	return ""
+}
+
+// generateTypes will generate no types for default examples, but customizations may
+// require their own defined types.
+func generateTypes(ex Example) string {
+	return ""
+}
+
+// buildShape will recursively build the referenced shape based on the json object
+// provided.
+func buildShape(ref *ShapeRef, shapes map[string]interface{}, isMap bool) string {
+	order := make([]string, len(shapes))
+	for k := range shapes {
+		order = append(order, k)
+	}
+	sort.Strings(order)
+
+	ret := ""
+	for _, name := range order {
+		shape := shapes[name]
+
+		// If the shape isn't a map, we want to export the value, since every field
+		// defined in our shapes are exported.
+		if len(name) > 0 && !isMap && strings.ToLower(name[0:1]) == name[0:1] {
+			name = strings.Title(name)
+		}
+
+		memName := name
+		if isMap {
+			memName = fmt.Sprintf("%q", memName)
+		} else if ref != nil {
+		}
+
+		switch v := shape.(type) {
+		case map[string]interface{}:
+			ret += buildComplex(name, memName, ref, v)
+		default:
+			ret += buildScalar(name, memName, ref, v)
+		}
+	}
+	return ret
+}
+
+func buildScalar(name, memName string, ref *ShapeRef, shape interface{}) string {
+	if ref == nil || ref.Shape == nil || ref.Shape.MemberRefs[name] == nil {
+		return ""
+	}
+
+	switch v := shape.(type) {
+	case bool:
+		//return fmt.Sprintf("%s: aws.Bool(%t),\n", memName, v)
+		return convertToCorrectType(memName, ref.Shape.MemberRefs[name].Shape.Type, fmt.Sprintf("%t", v))
+	case int:
+		if ref.Shape.MemberRefs[name].Shape.Type == "timestamp" {
+			return parseTimeString(ref, memName, fmt.Sprintf("%d", v))
+		} else {
+			return convertToCorrectType(memName, ref.Shape.MemberRefs[name].Shape.Type, fmt.Sprintf("%d", v))
+			//return fmt.Sprintf("%s: aws.Int64(%d),\n", memName, v)
+		}
+	case string:
+		t := ref.Shape.MemberRefs[name].Shape.Type
+		switch t {
+		case "timestamp":
+			return parseTimeString(ref, memName, fmt.Sprintf("%s", v))
+		case "blob":
+			if (ref.Shape.MemberRefs[name].Streaming || ref.Shape.MemberRefs[name].Shape.Streaming) && ref.Shape.Payload == name {
+				return fmt.Sprintf("%s: aws.ReadSeekCloser(bytes.NewBuffer([]byte(%q))),\n", memName, v)
+			} else {
+				return fmt.Sprintf("%s: []byte(%q),\n", memName, v)
+			}
+		default:
+			return convertToCorrectType(memName, t, v)
+		}
+	}
+	return ""
+}
+
+func convertToCorrectType(memName, t, v string) string {
+	switch t {
+	case "string":
+		return fmt.Sprintf("%s: aws.String(%q),\n", memName, v)
+	case "integer":
+		return fmt.Sprintf("%s: aws.Int64(%s),\n", memName, v)
+	case "boolean":
+		return fmt.Sprintf("%s: aws.Bool(%s),\n", memName, v)
+	default:
+		panic("Unsupported type: " + t)
+	}
+}
+
+func buildComplex(name, memName string, ref *ShapeRef, v map[string]interface{}) string {
```

Comment:
I noticed the `buildComplex` doesn't handle slices is that just because we didn't have any list data in example model?